deps/oob-1.0.2/jsonutil.c

```diff
@@ -404,7 +410,35 @@ void setoobConnectioninfo(const char* host,
     copyString("https", connectionInfo.protocol, 8);
   else
     copyString("http", connectionInfo.protocol, 8);
+}
 
+void setoobDsninfo(const char *key, const char *val) {
+    char* caps = calloc(strlen(key), sizeof(char));
+    upper(caps, key);
+
+    if (dsn == NULL) {
+        dsn = cJSON_CreateObject();
+    }
+    cJSON *value = cJSON_CreateString(val);
+    if (strcmp(caps, "PWD")) {
+        if (!strcmp(caps, "SERVER")) {
+            setdeployment(val);
+        }
+        cJSON_AddItemToObject(dsn, key, value);
```

Comment:
@sfc-gh-azhan who owns the memory for the string that is passed in to AddItemToObject ?

Reply:
The ODBC's m_connectionString owns the strings here. https://github.com/snowflakedb/snowflake-odbc/pull/83/files#diff-8a7d154ab6fecb24f34bcf3a036b2ddaR640
The `cJSON_AddItemToObject` will do a memcpy of the string so should be fine.